rest_framework_json_api/utils.py

```diff
@@ -368,6 +368,23 @@ def get_relation_instance(resource_instance, source, serializer):
     return True, relation_instance
 
 
+def get_serializer_from_context(context):
+    if hasattr(context["view"], "get_serializer"):
+        return context["view"].get_serializer()
+    elif hasattr(context["view"], "get_serializer_class"):
+        return context["view"].get_serializer_class()()
+    elif hasattr(context["view"], "serializer_class"):
+        return context["view"].serializer_class()
+
+
+def get_relationship_fields(fields):
+    return {
```

Comment:
As this is a public API I do not think how helpful this is as a utility on other cases. Actually in renderers.py the snippet ` isinstance(field, (relations.RelatedField, relations.ManyRelatedField)` is used twice. So if we had a method `is_relationship_field` those could be replaced and below in `format_drf_errors` you could do a simple dict comprehension.

In your code below you can then simply do a

Reply:
I agree! will do some refactors on this as well.